crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -148,22 +150,34 @@ pub fn update_frusta<T: Component + CameraProjection + Send + Sync + 'static>(
 }
 
 pub fn check_visibility(
+    mut thread_queues: Local<ThreadLocal<Cell<Vec<Entity>>>>,
     mut view_query: Query<(&mut VisibleEntities, &Frustum, Option<&RenderLayers>), With<Camera>>,
-    mut visible_entity_query: Query<(
-        Entity,
-        &Visibility,
-        &mut ComputedVisibility,
-        Option<&RenderLayers>,
-        Option<&Aabb>,
-        Option<&NoFrustumCulling>,
-        Option<&GlobalTransform>,
+    mut visible_entity_query: ParamSet<(
+        Query<&mut ComputedVisibility>,
+        Query<(
+            Entity,
+            &Visibility,
+            &mut ComputedVisibility,
+            Option<&RenderLayers>,
+            Option<&Aabb>,
+            Option<&NoFrustumCulling>,
+            Option<&GlobalTransform>,
+        )>,
     )>,
 ) {
+    // Reset the computed visibility to false
+    for mut computed_visibility in visible_entity_query.p0().iter_mut() {
+        computed_visibility.is_visible = false;
+    }
+
+    for queue in thread_queues.iter_mut() {
+        queue.get_mut().clear();
```

Comment:
Is this needed given that these are Local and you use append at the end which should anyway clear them out?

Reply:
Removed.